src/test/perf_local.cc

```diff
@@ -99,11 +99,11 @@ void discard(void* value) {
 double atomic_int_cmp()
 {
   int count = 1000000;
-  atomic_t value(11);
-  int test = 11;
+  std::atomic<unsigned> value = { 11 };
+  unsigned int test = 11;
   uint64_t start = Cycles::rdtsc();
   for (int i = 0; i < count; i++) {
-    value.compare_and_swap(test, test+2);
+    value.compare_exchange_weak(test, test+2);
```

Comment:
You're correct-- it would be fine if we only ran on architectures that provided cmp/swp instructions, but we can't promise that. I did a fixup PR for a previously-committed PR, and will patch it again here:
https://github.com/ceph/ceph/pull/15030#event-1076486762